code/hud/hudets.cpp

```diff
@@ -241,16 +248,16 @@ void ai_manage_ets(object* obj)
 		increase_recharge_rate(obj, ENGINES);
 	}
 
-	// emergency check
-	if (!(obj->flags[Object::Object_Flags::No_shields])) {
+	// emergency check for ships with shields
+	if (has_shields) {
```

Comment:
Whether or not the other place has_shields is kept, we may want to keep the original check here.  These tiny differences could cause AI differences that are difficult to detect later.  The original writer could have extended the check to include `!ship_p->ship_max_shield_strength`, but they didn't, and we just have to stick to retail as close as possible.

Reply:
Good point, I'll stick to exactly what the conditions were since the check in the early return looks at `!ship_p->ship_max_shield_strength`